datadog/dogstatsd.go

```diff
@@ -120,6 +120,10 @@ func (s *DogStatsdSink) AddSampleWithLabels(key []string, val float32, labels []
 	s.client.TimeInMilliseconds(flatKey, float64(val), tags, rate)
 }
 
+func (s *DogStatsdSink) Shutdown() {
+	s.client.Close()
```

Comment:
I may be wrong here, but this too can block https://github.com/DataDog/datadog-go/blob/v5.1.0/statsd/statsd.go#L698

This is fine, I'd just add docstrings around these new APIs and call that out.